File: w09/d03/enoch_riese/mail_objects_app/app/assets/javascripts/package.js
function Package(options) {
  this.contents = options.contents;
  this.fragile = options.fragile;
  this.broken = false;
}

Package.prototype.shake = function() {
  if (this.fragile === true) {
    this.broken = true;
  }
}
Package.prototype.isBroken = function() {
  return this.broken;
}

File: w09/d01/amy_ruan/mailer/lib/package.rb
require_relative 'mail'

class Package < Mail
  def initialize(options = {})
    @contents = []
    @ship_to_address = options[:ship_to_address]
    @is_fragile = options[:is_fragile]
    @is_broken = false
  end

  def contents
    @contents
  end

  def is_fragile?
    @is_fragile
  end

  def is_broken?
    @is_broken
  end

  def shake
    @is_broken = true if @is_fragile
  end

end